### DISCUSSION TITLE
File Type - CDE 11416926

### DISCUSSION THREAD

**Discussion by GeoffLyle** [CBIIT/ccdi-federation-api#53] [Metadata Field Ratification]
caDSR Link: [11416926](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=11416926%20and%20ver_nr=1)

Definition: A defined organization or layout representing and structuring data in a computer file.

Discussion: Still needs review of permissible values and definitions.

**Comment by Ssandor13** [comment]
St. Jude is able to map our permissible values to this field, however, we do want to discuss with the Federation the best path forward for mapping our genomic VCF file types. Does mapping to VCF suffice? We'd love to hear from the Federation how they are mapping their genomic VCF files to this CDE. 

Further, our somatic VCF files would be mapped to the VCF permissible value, but does this pose any assumptions or risks?

**Reply by hbeale** [reply]
I anticipate that we're more likely to use MAF than VCF. We often only have MAF-level data, not the levels of detail in VCFs.

**Comment by hbeale** [comment]
This file type CDE works for Treehouse, and generally looks solid across all the permissible values I'm familiar with. We anticipate first using TSV.

**Comment by GeoffLyle** [comment]
CDE accepted per January 25th meeting.

Concerns about genomic vs. somatic VCF (gVCF vs. VCF) are not resolved by this CDE alone.  This information may be found in the file name (#52) or comparing 'Tumor' and 'Normal' in the Sample Tumor Status field (#39)

**Reply by Ssandor13** [reply]
After further consideration, our team would like to propose to extend this CDE by incorporating a new permissible value specifically designed to accommodate genomic Variant Call Format (gVCF) independently. While the format meets VCF specifications, gVCF encompasses records for all genomic sites, regardless of whether a variant call is present or not. This is to facilitate joint analysis of a cohort in subsequent analytical steps, thus the utility of this file type may vary by users and we should denote this as a different file type than a regular VCF file type. I'd also like to clarify that while the ".g.vcf" extension is presented in the file name, it may not be apparent to the user to understand what file type this really is.

**Reply by hbeale** [reply]
@Ssandor13 I see the value of having distinct gVCF and VCF CDEs. Do people often suffix these files with ".gvcf", or are the consistently the same as your example, ".g.vcf"?

**Reply by GeoffLyle** [reply]
@Ssandor13 
Would you be willing to propose a new Permissible Value in the same format as the other PVs in this CDE ('Permissible Value', 'VM Long Name', 'VM Definition').  After discussing gVCF v. VCF formats with Holly, I agree that users would likely want to know the difference between these file types.
@claymcleod Would it possible to implement this CDE with this caveat in the API.  Similar to our proposed implementation of #47 - Library Selection Method - where we would like to either change the definition of "PolyA" or add a new Permissible Value "PolyA Selection".

**Reply by claymcleod** [reply]
FWIW, there is no convention I am aware of in the community regarding `.g.vcf` and `.gvcf`. It's in this middle ground of a file type where it is pervasive enough to warrant its own file type under this CDE but not enough that there are strong community conventions IMO.

Geoff, I haven't taken a look at #47 yet, but it might be. I'll have to look at the details. I have a slight preference for only including permissible values in the CDE and then just updating the code when the permissible values we want are added.

**Reply by Ssandor13** [reply]
**Proposal:** 

'Permissible Value': gVCF, 'VM Long Name': Genomic VCF, 'VM Definition': A text-based electronic file used for storing sequence pileup and allele count data at all genomic sites. In contrast to VCF files, gVCF files make no assertion regarding called variants. Instead, their purpose is to cache gathered information that is used by the GATK's Haplotype Caller to avoid reprocessing the entire cohort when a single, new sample is added. These files conform to the VCF file format specification.

**Reply by Ssandor13** [reply]
@GeoffLyle I wanted to circle back to this and see what the consensus is since we are ready to implement this CDE into our API and map our permissible values? We will, in the meantime, map both to VCF but still propose to have a new permissible value added to this CDE.

**Reply by GeoffLyle** [reply]
@jknable 

Are we able to add a 'gVCF' permissible value to this CDE:

Copied @Ssandor13 proposal:

'Permissible Value': gVCF 
'VM Long Name': Genomic VCF 
'VM Definition': A text-based electronic file used for storing sequence pileup and allele count data at all genomic sites. In contrast to VCF files, gVCF files make no assertion regarding called variants. Instead, their purpose is to cache gathered information that is used by the GATK's Haplotype Caller to avoid reprocessing the entire cohort when a single, new sample is added. These files conform to the VCF file format specification.

**Reply by jknable** [reply]
@GeoffLyle we are working with the NCI terminology group to get a controlled term created for 'gVCF' as well as the CDE owner to see about adding it to the existing CDE or if a new CDEs will be needed. Should have an answer by early next week.

**Reply by jknable** [reply]
@GeoffLyle @Ssandor13 @jknable
We have added a permissible value of "gVCF" to this CDE.  The terminology team created the following term and associated definition.  The definition and additional synonyms are based on information from the EDAM (Bioscientific data analysis ontology) standard provided by the CDE owner. 

NCI Concept: Genomic Variant Call Format
NCI Code: C204352
Synonym: GVCF; Genomic VCF; g.vcf.gz; g.vcf
Definition: A text file format used for storing gene sequence variations. It is a type of variant call format that has a record for all sites, whether there is a variant call there or not. The format includes metrics of the confidence that positions are actually non-variant vs. a factor of minimum read-depth and genotype quality.

We added your suggested term definition above as an alternate VM definition for "gVCF" which can be accessed by clicking the "VM Alternate Definition" node in the tool tree.

**Comment by Ssandor13** [comment]
@GeoffLyle @hbeale I wanted to kindly share thoughts here as I reached directly to Holly a few weeks ago about the idea to add a PV for HTSeq Counts, however, is that being too specific? Right now, we are mapping that to the txt permissible value.

**Reply by Ssandor13** [reply]
@GeoffLyle @hbeale @jknable 
St. Jude discussed internally and we wanted to propose the following and see what others thought about this?

**Permissible Value:** HTSeq Count
**VM Long Name:** HTSeq Count Format
**VM Definition:** "A file format generated by HTSeq Count."

**Reply by jknable** [reply]
Based on the term "count" in your proposed permissible value, is the data to be collected the "count of how many reads map to a given genomic feature" in a given aligned sequencing file?  If so, this permissible value seems like it would be a result/output of a function/procedure performed on a data file (ex BAM/SAM file), not the actual format of file.  Am I correct with this assumption?

**Reply by Ssandor13** [reply]
@jknable I apologize, I unfortunately stripped information that should have been included in the proposed VM definition. Please see our updated proposal below:

**Permissible Value:** HTSeq Count
**VM Long Name:** HTSeq Count Format
**VM Definition:** "A file format generated by HTSeq Count that represents the quantification results of mapped sequencing reads against genomic features, typically genes. It contains two main columns Gene ID and Count."

**Reply by jknable** [reply]
@GeoffLyle @hbeale
For consistency with our NCI terminologies, the NCI EVS terminology specialists have recommended this long name and definition.  Will this work?

VM Long Name: HTSeq-Count Output Format
VM Definition: A tab-delimited file that represents the number of reads obtained during high-throughput sequencing (HTSeq) that map to known genomic sequences. The data is represented in two columns, one containing an identifier for a genomic feature, which can be a database-supplied identifier or a gene symbol, followed by a second column containing the number of mapped reads.

**Reply by hbeale** [reply]
We don't have HTSeq Count files, so I'm not an expert, but I'm don't see any problem with this.

**Reply by mcrusch** [reply]
Does anybody know if this file format is produced by anything other than HTSeq Count?

Looking over the File Type CDE, most of the file types are fairly open/generic, so the value, long name, and definition are built around the content or formatting of the file and don't reference specific tools.  There are a few (such as "tar" and "Thermo raw") that are tool-specific, and so the value, name, and definition are built around that tool.  I think these are good decisions, and our approach to this value should continue in that vein.

Also, I've never heard anybody refer to high-throughput sequencing generically as HTSeq.  I think that's only used as the proper name of that one tool.

So, assuming this format is specific to the HTSeq tool, then I would propose the following definition, which incorporates elements of the previous two suggestions: "A tab-delimited file generated by HTSeq Count that represents the number of reads that map to known genomic sequences. The data is represented in two columns, one containing an identifier for a genomic feature, which can be a database-supplied identifier or a gene symbol, followed by a second column containing the number of mapped reads."

If this is a generic format, then I would drop all reference to "HTSeq Count".  I don't think this is the case, however.

**Reply by jknable** [reply]
@GeoffLyle
For consistency with our NCI terminologies, the NCI EVS terminology specialists reviewed your recommendation and have revised the definition as follows:

**NCI DEF:** A tab-delimited file generated by the HTSeq Count software tool. The file represents the number of reads obtained during high-throughput sequencing that map to known genomic sequences. The data is represented in two columns, one containing an identifier for a genomic feature, which can be a database-supplied identifier or a gene symbol, followed by a second column containing the number of mapped reads.

**Reply by GeoffLyle** [reply]
@Ssandor13 @mcrusch 
How does this definition look to you?  Looks a good rewording of the original definition you provided.

**Reply by Ssandor13** [reply]
@GeoffLyle and @jknable  We agree with this proposed definition. Thank you!

**Reply by jknable** [reply]
@GeoffLyle and @jknable  Thank you, we have asked the terminology team to finalize this def/term.  Once that is completed, we will update the CDE.

**Reply by Ssandor13** [reply]
@jknable I wanted to kindly follow up on this to see whether this will be updated for v1? Thanks!

**Reply by jknable** [reply]
We received permission today to add that value as a file type.  It should be visible in the CDE by COB Friday.
